az-hadoop-jobtype-plugin/src/main/java/azkaban/jobtype/HadoopTuningSecurePigWrapper.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2018 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.jobtype;
+
+import static org.apache.hadoop.security.UserGroupInformation.HADOOP_TOKEN_FILE_LOCATION;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.security.PrivilegedExceptionAction;
+import java.util.Properties;
+
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.log4j.Logger;
+import org.apache.pig.tools.pigstats.PigStats;
+
+import azkaban.jobtype.pig.PigUtil;
+import azkaban.jobtype.tuning.TuningCommonConstants;
+import azkaban.jobtype.tuning.TuningErrorDetector;
+import azkaban.jobtype.tuning.TuningException;
+import azkaban.jobtype.tuning.TuningParameterUtils;
+import azkaban.utils.Props;
+
+
+/**
+ * This class represent wrapper for running pig job with tuning enabled.
+ */
+public class HadoopTuningSecurePigWrapper {
+
+  public static final String WORKING_DIR = "working.dir";
+
+  private static File pigLogFile;
+
+  private static Props props;
+
+  private static final Logger logger;
+
+  /**
+   * In case if job is failed because of auto tuning parameters, it will be retried with the best parameters
+   * we have seen so far. Maximum number of try by default is 2, which can be configured using parameter
+   * tuning.job.retry.count
+   */
+
+  private static final String TUNING_JOB_RETRY_COUNT = "tuning.job.retry.count";
+
+  static {
+    logger = Logger.getRootLogger();
+  }
+
+  /**
+   * This function runs the pig job with tuning enabled. In case tuning job fails in first try, it asks for
+   * last best parameter from tuning and re-run the job. In case tuning end point is not responding this job
+   * will run with default parameters.
+   * @param args
```

Comment:
Should have an empty line before `@param`

Reply:
Changed.